### PR TITLE
Added Progress Bar and Enroll Butto in the Project Section

### PR DESCRIPTION
**Pull request by BunnyBharath91**
-Added a Progress Bar in the Project Section 
-Added an Enroll Button in the Project Section

### CODE REVIEW

File: src/database/query/project.ts [hunk 1]
```diff
@@ -401,6 +418,110 @@ const deleteChapterFromSectionInDB = async ({
...
+      (chapter) => chapter.chapterId.toString() === chapterId.toString()
+    );
+
+    if (chapterIndex === -1) {
+      // If chapter is not found, add it with the given status
+      userProject.sections[sectionIndex].chapters.push({
+        chapterId: chapterId,
+        isCompleted: isCompleted,
+        createdAt: new Date(),
+        updatedAt: new Date(),
```
Comment (imsks) [inline src/database/query/project.ts]: These two timestamps gets updated Automatically if your DB model is having timestamp=true prop

File: src/database/query/project.ts [hunk 2]
```diff
@@ -401,6 +418,110 @@ const deleteChapterFromSectionInDB = async ({
...
+
+    const sections = project.sections.map((section: any) => ({
+      sectionId: section.sectionId,
+      createdAt: new Date().toISOString(),
+      updatedAt: new Date().toISOString(),
+      chapters: section.chapters.map((chapter: any) => ({
+        chapterId: chapter.chapterId,
+        isCompleted: false,
+        createdAt: new Date().toISOString(),
+        updatedAt: new Date().toISOString(),
```
Comment (imsks) [inline src/database/query/project.ts]: same here

File: src/database/query/project.ts [hunk 3]
```diff
@@ -401,6 +418,110 @@ const deleteChapterFromSectionInDB = async ({
...
+  try {
+    const project = await Project.findById(projectId).lean();
+    if (!project) {
+      return { error: 'Project not found' };
+    }
+
+    const sections = project.sections.map((section: any) => ({
+      sectionId: section.sectionId,
+      createdAt: new Date().toISOString(),
+      updatedAt: new Date().toISOString(),
```
Comment (imsks) [inline src/database/query/project.ts]: same here

File: src/database/query/project.ts [hunk 4]
```diff
@@ -401,6 +418,110 @@ const deleteChapterFromSectionInDB = async ({
...
+      })),
+    }));
+
+    // Create user project data with the initialized sections
+    const userProject = await UserProject.create({
+      userId,
+      projectId,
+      sections,
+      createdAt: new Date(),
+      updatedAt: new Date(),
```
Comment (imsks) [inline src/database/query/project.ts]: here also

**Review by imsks — CHANGES_REQUESTED** [review]

File: src/database/query/project.ts (hunk 1 above)
Comment (BunnyBharath91) [reply src/database/query/project.ts]: resolved

**Comment by BunnyBharath91** [comment]
removed successfully

**Review by imsks — APPROVED** [review]

**Pull request merged**